jvb/src/test/kotlin/org/jitsi/videobridge/cc/vp9/Vp9AdaptiveSourceProjectionTest.kt

```diff
@@ -1010,13 +1010,172 @@ class Vp9AdaptiveSourceProjectionTest {
+                val accepted = context.accept(
+                    packetInfo,
+                    getIndex(0, packet.spatialLayerIndex, packet.temporalLayerIndex), targetIndex
+                )
+                Assert.assertFalse(accepted)
+                if (packet.isEndOfPicture) {
+                    expectedTs = RtpUtils.applyTimestampDelta(expectedTs, 3000)
+                }
+            }
+
+            /* Request a keyframe.  Will be sent as of the next frame. */
+            generator.requestKeyframe()
+            while (generator.packetOfFrame != 0) {
+                packetInfo = generator.nextPacket()
+                packet = packetInfo.packetAs()
+                val accepted = context.accept(
+                    packetInfo,
+                    getIndex(0, packet.spatialLayerIndex, packet.temporalLayerIndex), targetIndex
+                )
+                Assert.assertFalse(accepted)
+                if (packet.isEndOfPicture) {
+                    expectedTs = RtpUtils.applyTimestampDelta(expectedTs, 3000)
+                }
+            }
+
+            for (i in 0..999) {
+                packetInfo = generator.nextPacket()
+                packet = packetInfo.packetAs()
+                val accepted = context.accept(
+                    packetInfo,
+                    getIndex(0, packet.spatialLayerIndex, packet.temporalLayerIndex), targetIndex
+                )
+                if (packet.isStartOfFrame && packet.temporalLayerIndex == 0) {
+                    expectedTl0PicIdx = applyTl0PicIdxDelta(expectedTl0PicIdx, 1)
+                }
+                val endOfPicture = packet.isEndOfPicture
+                if (packet.temporalLayerIndex <= targetTid &&
+                    (
+                        packet.spatialLayerIndex == targetSid ||
+                            (packet.isUpperLevelReference && packet.spatialLayerIndex < targetSid)
+                        )
+                ) {
+                    Assert.assertTrue(accepted)
+                    context.rewriteRtp(packetInfo)
+                    Assert.assertEquals(expectedSeq, packet.sequenceNumber)
+                    Assert.assertEquals(expectedTs, packet.timestamp)
+                    Assert.assertEquals(expectedPicId, packet.pictureId)
+                    Assert.assertEquals(expectedTl0PicIdx, packet.TL0PICIDX)
+                    expectedSeq = RtpUtils.applySequenceNumberDelta(expectedSeq, 1)
+                } else {
+                    Assert.assertFalse(accepted)
+                }
+                if (endOfPicture) {
+                    expectedTs = RtpUtils.applyTimestampDelta(expectedTs, 3000)
+                    expectedPicId = applyExtendedPictureIdDelta(expectedPicId, 1)
+                }
+            }
+            suspended *= 2
+        }
+    }
+
+    @Test
+    fun sourceSuspensionTest() {
+        val generator = ScalableVp9PacketGenerator(1)
+        runSourceSuspensionTest(generator, 2)
```

Comment:
nit: I think `getIndex(0, 0, 2)` would be easier to understand (or `getIndex(tid=2)` if you were to add default values to `getIndex`).